### PR TITLE
mcs: speed up and stablize tests

### LINKED ISSUE #7399: TestForwardStoreHeartbeat is unstable
### Which jobs are failing
```
2023-11-21T04:23:45.0314918Z     testutil.go:64: 
2023-11-21T04:23:45.0315360Z         	Error Trace:	/home/runner/work/pd/pd/pkg/utils/testutil/testutil.go:64
2023-11-21T04:23:45.0316148Z         	            				/home/runner/work/pd/pd/tests/integrations/mcs/scheduling/server_test.go:187
2023-11-21T04:23:45.0316488Z         	Error:      	Condition never satisfied
2023-11-21T04:23:45.0316805Z         	Test:       	TestServerTestSuite/TestForwardStoreHeartbeat
```

### CI link
https://github.com/tikv/pd/actions/runs/6939184278/job/18876116632?pr=7398

### PR DESCRIPTION
**Pull request by rleungx**
### What problem does this PR solve?

Issue Number: Close #7399.

### What is changed and how does it work?

```commit-message
```

### Check List

Tests

- Unit test

### Release note

```release-note
None.
```

### CODE REVIEW

**Review by lhy1024 — APPROVED** [review]

File: tests/integrations/mcs/scheduling/server_test.go [hunk 1]
```diff
@@ -96,6 +96,10 @@ func (suite *serverTestSuite) TestAllocID() {
 func (suite *serverTestSuite) TestAllocIDAfterLeaderChange() {
 	re := suite.Require()
 	re.NoError(failpoint.Enable("github.com/tikv/pd/pkg/mcs/scheduling/server/fastUpdateMember", `return(true)`))
+	pd2, err := suite.cluster.Join(suite.ctx)
```
Comment (CabinfeverB) [inline tests/integrations/mcs/scheduling/server_test.go:99]: Why modify this?

**Review by CabinfeverB — APPROVED** [review]
rest LGTM

File: tests/integrations/mcs/scheduling/server_test.go (hunk 1 above)
Comment (rleungx) [reply tests/integrations/mcs/scheduling/server_test.go:99]: previous:
```
--- PASS: TestServerTestSuite (35.10s)
    --- PASS: TestServerTestSuite/TestAllocID (0.29s)
    --- PASS: TestServerTestSuite/TestAllocIDAfterLeaderChange (1.83s)
    --- PASS: TestServerTestSuite/TestDynamicSwitch (0.77s)
    --- PASS: TestServerTestSuite/TestForwardRegionHeartbeat (5.19s)
    --- PASS: TestServerTestSuite/TestForwardStoreHeartbeat (0.19s)
    --- PASS: TestServerTestSuite/TestPrimaryChange (0.37s)
    --- PASS: TestServerTestSuite/TestSchedulerSync (3.94s)
    --- PASS: TestServerTestSuite/TestStoreLimit (5.71s)
```
after:
```
--- PASS: TestServerTestSuite (27.96s)
    --- PASS: TestServerTestSuite/TestAllocID (0.29s)
    --- PASS: TestServerTestSuite/TestAllocIDAfterLeaderChange (10.34s)
    --- PASS: TestServerTestSuite/TestDynamicSwitch (0.78s)
    --- PASS: TestServerTestSuite/TestForwardRegionHeartbeat (5.20s)
    --- PASS: TestServerTestSuite/TestForwardStoreHeartbeat (0.19s)
    --- PASS: TestServerTestSuite/TestPrimaryChange (0.39s)
    --- PASS: TestServerTestSuite/TestSchedulerSync (3.95s)
    --- PASS: TestServerTestSuite/TestStoreLimit (5.72s)
```

**Pull request merged**